tests/Configuration.UnitTests/Support/ConfigurationExample.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace Microsoft.Omex.Extensions.Configuration.UnitTests.Support
+{
+	public class ConfigurationExample
+	{
+		[Required(AllowEmptyStrings = false)]
+		public string? ValidatedConfiguration { get; set; }
```

Comment:
Why is it nullable when its Required?

Reply:
Unfortuntely, the data annotations do not have any influence on the compiler warning (in our case, errors) triggered when it recognises possible nullable reference. Hence, the alternatives are two:
- Declaring the property as nullable.
- Declaring the property as not nullable and equalise it to a predefined value.

Reply:
I would go with the default value to make it more visible that it shouldn't be null but feel free to ignore that

Reply:
You're right, expressing the type as a non-nullable string is semantically correct. Thanks!